### PR TITLE
IGNITE-21141 MvccProcessor removal

### PR DESCRIPTION
**Pull request by shishkovilja**
`MvccProcessor` and all corresponding code was removed.

TC bot run: https://tcbot2.sbt-ignite-dev.ru/pr.html?serverId=apache&suiteId=IgniteTests24Java8_RunAll&branchForTc=pull/11128/head&action=Latest 

----
Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [x] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [x] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxAbstractEnlistFuture.java [hunk 1]
```diff
@@ -918,11 +917,7 @@ private boolean skipNearLocalUpdate(ClusterNode node, boolean moving) {
      * @throws ClusterTopologyCheckedException If failed.
      */
     private void checkCoordinatorVersion() throws ClusterTopologyCheckedException {
```
Comment (anton-vinogradov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxAbstractEnlistFuture.java:919]: Any reason to keep this empty method?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java [hunk 2]
```diff
@@ -289,13 +281,12 @@ private void onAffinityReady(@NotNull IgniteInternalFuture<AffinityTopologyVersi
 
     /** */
     private void ackQueryDone(MvccSnapshot snapshot) {
-        if (snapshot != null)
-            cctx.shared().coordinators().ackQueryDone(snapshot, id);
+        // No-op.
```
Comment (anton-vinogradov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java:284]: Any reason to keep this empty method?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java [hunk 3]
```diff
@@ -289,13 +281,12 @@ private void onAffinityReady(@NotNull IgniteInternalFuture<AffinityTopologyVersi
...
     }
 
     /** */
     @NotNull private MvccCoordinator coordinator() {
```
Comment (anton-vinogradov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java:288]: Any reason to keep this empty method?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java [hunk 4]
```diff
@@ -2740,19 +2739,7 @@ private boolean haveSystemTxForThread(long threadId) {
...
-
-        cctx.database().checkpointReadLock();
-
-        try {
-            cctx.coordinators().updateState(tx.mvccSnapshot(), state0, tx.local());
-        }
-        finally {
-            cctx.database().checkpointReadUnlock();
-        }
+        // No-op.
```
Comment (anton-vinogradov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java:2742]: Any reason to keep this empty method?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java [hunk 5]
```diff
@@ -2774,10 +2761,7 @@ private byte toMvccState(TransactionState state) {
      *  @param tx Transaction.
      */
     public void mvccFinish(IgniteTxAdapter tx) {
```
Comment (anton-vinogradov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java:2763]: Any reason to keep this empty method?

**Review by anton-vinogradov — CHANGES_REQUESTED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxAbstractEnlistFuture.java (hunk 1 above)
Comment (shishkovilja) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxAbstractEnlistFuture.java]: Good point! I have removed this method.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java (hunk 2 above)
Comment (shishkovilja) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java]: Good point! I have removed this method.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java (hunk 3 above)
Comment (shishkovilja) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/mvcc/MvccQueryTrackerImpl.java:279]: Good point! I have removed this method.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java (hunk 4 above)
Comment (shishkovilja) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java]: Good point! I have removed this method.

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java (hunk 5 above)
Comment (shishkovilja) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java]: Good point! I have removed this method.

**Pull request merged**